tests/integration/conftest.py

```diff
@@ -54,17 +70,78 @@ async def unit_ip_list(get_unit_ip_list):
 
 
 @pytest_asyncio.fixture(scope="module")
-async def app(ops_test: pytest_operator.plugin.OpsTest, content_cache_image: str):
+async def nginx_integrator_app(ops_test: OpsTest):
+    nginx_integrator_app_name = "nginx-ingress-integrator"
+    nginx_integrator_app = await ops_test.model.deploy(nginx_integrator_app_name, trust=True)
+    await ops_test.model.wait_for_idle()
+    assert (
+        ops_test.model.applications[nginx_integrator_app_name].units[0].workload_status
+        == ActiveStatus.name
+    )
+    yield nginx_integrator_app
+
+
+@pytest_asyncio.fixture(scope="module")
+async def app(
+    ops_test: OpsTest, app_name: str, content_cache_image: str, nginx_integrator_app: Application
+):
     """Content-cache-k8s charm used for integration testing.
 
-    Builds the charm and deploys it for testing purposes.
+    Deploy kubecon charm, builds the charm and deploys it for testing purposes.
     """
-    my_charm = await ops_test.build_charm(".")
+    hello_kubecon_app_name = "hello-kubecon"
+    ops_test.model.deploy(hello_kubecon_app_name)
+    await ops_test.model.deploy(hello_kubecon_app_name)
+    await ops_test.model.wait_for_idle()
+
+    app_charm = await ops_test.build_charm(".")
     application = await ops_test.model.deploy(
-        my_charm,
+        app_charm,
+        application_name=app_name,
         resources={"content-cache-image": content_cache_image},
         series="jammy",
     )
-    await ops_test.model.wait_for_idle()
+
+    try:
+        await ops_test.model.wait_for_idle(raise_on_blocked=True)
```

Comment:
Would `await ops_test.model.wait_for_idle(status=<blocked status>)` work here?

Reply:
Actually, this is something that I couldn't figure out how to explain. My first try was precisely this, wait for blocked status, but it didn't work locally. So, I used this to ensure that the app is ready for the relation.

Reply:
It doesn't really make sense that there is a difference, maybe it is worth filing a bug for this on `pytest-operator`/ `libjuju`?